src/arrayMethodJoin.js

```diff
@@ -5,7 +5,25 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    const actualSeparator = separator === undefined ? ',' : separator;
+
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      const element = this[i];
+
+      if (i > 0) {
+        result += actualSeparator;
+      }
+
+      if (element === null || element === undefined) {
+        result += '';
```

Comment:
There is no need to append an empty string to the result. This operation could be removed for performance and readability.